peps/pep-0759.rst

```diff
@@ -0,0 +1,472 @@
+
+:pep:`708` provides additional background about dependency confusion attacks,
+and takes a different approach to preventing them. At its core, PEP 708 allows
+repository owners to indicate that projects track across different
+repositories, which allows installers to determine how to treat the global
+package namespace when combined across multiple repositories. PEP 708 has been
+provisionally accepted, pending several required conditions as outlined in PEP
+708, some of which may have an indeterminate future. As PEP 708 itself says,
+this won't by itself solve dependency confusion attacks, but is one way to
+provide enough information to installers to help minimize these attacks.
+
+While there can still be valid use cases for standing up a totally independent
+package index (such as providing richer platform support for GPUs until a
+fully formed `variant proposal
+<https://discuss.python.org/t/selecting-variant-wheels-according-to-a-semi-static-specification/53446>`__
+is accepted), this PEP takes a different, simpler approach and doesn't replace
+any of the existing, proposed, or approved package index cooperation
+specifications.
+
+This PEP also preserves the core purpose of PyPI, and allows it to
+remain the traditional, canonical, centralized index of all Python
+packages.
+
+It's not just about indexes
+===========================
+
+The problem *this* PEP solves is not just reduction or elimination of
+dependency confusion attacks. There is another class of problem that a safe
+external wheel hosting solution can mitigate: artifact size limits.
+
+PyPI imposes a `default artifact size limit <https://pypi.org/help/#file-size-limit>`__ of
+100MiB and a default overall `project size limit
+<https://pypi.org/help/#project-size-limit>`__ of 10GiB. Most packages and artifacts can
+easily fit in these limits, even for packages containing binary extension modules for a
+variety of platforms. A small, but important class of packages routinely exceed these
+limits, requiring them to submit PyPI support tickets `requesting higher limits`_ It's not
+necessarily difficult to get resolution on such exceptions, but it is a special process
+that can take some time to resolve, and the criteria for granting such exceptions isn't
+well documented.
+
+Reduce operational complexity
+=============================
+
+Standing up and maintaining an entire package index can be a complex
+operational solution, both time and resource intensive. This is especially
+true if the main purpose of such an index is just to avoid file size
+limitations. The external index approach also imposes a tricky UX on consumers
+of projects on the external index, requiring them to understand how CLI
+options such as ``--external-index-url`` work, along with the security
+implications of such flags. It would be much easier for both producers and
+consumers of large wheel packages to just stand up and maintain a simple web
```

Comment:
```suggestion
consumers of large wheel packages to just set up and maintain a simple web
```
Is "stand up" just an idiom I don't know about?

Reply:
Perhaps!  "Standing up a service" but if it's not clear to you it likely won't be clear to others, so let's go with "set[ting] up" as per your suggestion.  Thanks!.

Reply:
"standing up" a server means to set it up. I think it is definitely better to use set up to be clearer to a wide audience though, thanks for catching that!